corehq/apps/app_manager/README.md

```diff
@@ -1,21 +1,57 @@
-# Application terminology
+# App manager terminology
 
-For any given application, there are a number of different documents.
+## Applications (and builds)
 
-The primary application document is an instance of `Application`.  This
-document's id is what you'll see in the URL on most app manager pages. Primary
-application documents should have `copy_of == None` and `is_released ==
-False`. When an application is saved, the field `version` is incremented.
+An application typically has many different `Application` documents: 
+one for the current/primary/canonical application, plus one for each saved build.
+The current app's id is what you'll see in the URL on most app manager pages;
+most pages will redirect to the current app if given the id of an older build.
+
+In saved builds, `copy_of` contains the primary app's id, while it's `None` for the primary app itself.
+The property `master_id` gets the id of the primary app doc, whether it's called on a saved build or on the primary app.
+
+Within code, "build" should always refer to a saved build,
+but "app" is used for both the current app and saved builds.
+The ambiguity of "app" is occasionally a source of bugs.
```

Comment:
and should be fixed where found?

Reply:
Probably, although this ambiguity is all over the place and it's often difficult to decide how to rename - like, `get_app` gets both apps and builds, should we call it `get_app` or `get_app_or_build`, or have both a `get_app` and a `get_build`?

Reply:
i think that method let's the user not worry about what they need and just give an `id` so `get_app_or_build` sounds right.
However i think this line was talking about variables used but better method names would be good too.